packages/composites/glv-token/src/transport/price.ts

```diff
@@ -0,0 +1,377 @@
+      const symbol = mapSymbol(glv.markets[m].indexToken, this.tokensMap).symbol
+      indexTokensPrices.push([priceResult.prices[symbol].ask, priceResult.prices[symbol].bid])
+    })
+
+    const glvTokenPriceContractParams = [
+      this.settings.DATASTORE_CONTRACT_ADDRESS,
+      glvInfo.markets,
+      indexTokensPrices,
+      [priceResult.prices[glv.longToken.symbol].ask, priceResult.prices[glv.longToken.symbol].bid],
+      [
+        priceResult.prices[glv.shortToken.symbol].ask,
+        priceResult.prices[glv.shortToken.symbol].bid,
+      ],
+      glv_address,
+    ]
+
+    // Prices have a spread from min to max. The last param (maximize-true/false) decides whether to maximize the market token price
+    // or not. We get both values and return the median.
+
+    const [[maximizedValue], [minimizedValue]] = await Promise.all([
+      this.glvReaderContract.getGlvTokenPrice(...glvTokenPriceContractParams, true),
+      this.glvReaderContract.getGlvTokenPrice(...glvTokenPriceContractParams, false),
+    ])
+
+    const maximizedPrice = Number(utils.formatUnits(maximizedValue, SIGNED_PRICE_DECIMALS))
+    const minimizedPrice = Number(utils.formatUnits(minimizedValue, SIGNED_PRICE_DECIMALS))
+    const result = median([minimizedPrice, maximizedPrice])
+
+    return {
+      data: {
+        result: result,
+        sources: priceResult.sources,
+      },
+      statusCode: 200,
+      result: result,
+      timestamps: {
+        providerDataRequestedUnixMs,
+        providerDataReceivedUnixMs: Date.now(),
+        providerIndicatedTimeUnixMs: undefined,
+      },
+    }
+  }
+
+  // Fetches the lwba price info from multiple source EAs, calculates the median for bids and asks per asset and fixes the price precision
+  private async fetchPrices(assets: string[], dataRequestedTimestamp: number) {
+    // priceData holds raw bid/ask values per asset from source EAs response
+    const priceData = {} as PriceData
+
+    const sources = [
+      { url: this.settings.TIINGO_ADAPTER_URL, name: 'tiingo' },
+      { url: this.settings.COINMETRICS_ADAPTER_URL, name: 'coinmetrics' },
+      { url: this.settings.NCFX_ADAPTER_URL, name: 'ncfx' },
+    ]
+
+    //priceProviders contains assets with a list of sources where asset price was successfully fetched
+    const priceProviders: Record<string, string[]> = {}
+
+    const promises = []
+
+    for (let i = 0; i < sources.length; i++) {
+      const source = sources[i]
+      const assetPromises = assets.map(async (asset) => {
+        const base = this.unwrapAsset(asset)
+        const requestConfig = {
+          url: source.url,
+          method: 'POST',
+          data: {
+            data: {
+              endpoint: 'crypto-lwba',
+              base,
+              quote: 'USD',
+            },
+          },
+        }
+
+        if (unsupportedAssets[source.name].includes(asset)) {
+          // skip assets not supported by EA to avoid Rate Limiting Error
+          undefined
```

Comment:
??

Reply:
I don't really like this idea. We should always try to get the price from all sources and  `MIN_REQUIRED_SOURCE_SUCCESS` is just for some rare cases where DP is unavailable. Deliberately skipping some assets will cause issues sooner or later.